variant-remapping-automation/prepare_remapping_table.py

```diff
@@ -0,0 +1,246 @@
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query, execute_query
+from ebi_eva_common_pyutils.taxonomy.taxonomy import get_scientific_name_from_ensembl
+
+from genome_target_tracker import get_tax_latest_asm_from_eva, check_supported_target_assembly
+
+logging_config.add_stdout_handler()
+logger = logging_config.get_logger(__name__)
+
+
+def prepare_remapping_table(private_config_xml_file, remapping_version):
+    eva_assemblies = get_eva_asm_tax(args.private_config_xml_file)
+    dbsnp_assemblies = get_dbsnp_asm_tax()
+    scientific_names_from_release_table = get_scientific_name_from_release_table(args.private_config_xml_file)
+    tax_latest_support_asm = get_tax_latest_asm_from_eva(private_config_xml_file)
+    tax_all_supported_asm = get_tax_all_supported_assembly_from_eva(private_config_xml_file)
+    asm_studies_acc_after_remapping = get_studies_for_remapping(private_config_xml_file)
+    taxonomy_with_mismatch_assembly, _, _ = check_supported_target_assembly(private_config_xml_file)
+
+    insert_entries_for_new_assembly_found_in_ensembl(private_config_xml_file, remapping_version, eva_assemblies,
+                                                     dbsnp_assemblies, scientific_names_from_release_table,
+                                                     tax_all_supported_asm, taxonomy_with_mismatch_assembly)
+
+    insert_entries_for_new_studies(private_config_xml_file, remapping_version, eva_assemblies, dbsnp_assemblies,
+                                   scientific_names_from_release_table, tax_latest_support_asm, tax_all_supported_asm,
+                                   taxonomy_with_mismatch_assembly, asm_studies_acc_after_remapping)
+
+
+def insert_entries_for_new_assembly_found_in_ensembl(private_config_xml_file, remapping_version, eva_assemblies,
+                                                     dbsnp_assemblies, scientific_names_from_release_table,
+                                                     tax_all_supported_asm, taxonomy_with_mismatch_assembly):
+    # Given a taxonomy (tax1) with assemblies(asm1, asm2, asm3) if ensemble has a new assembly (tax1->asm4)
+    # remap all assemblies to asm4.  {asm1 -> asm4, asm2-> asm4, asm3 -> asm4}
+    release_version = args.remapping_version + 2
+    for tax_id, diff_asm in taxonomy_with_mismatch_assembly.items():
+        scientific_name = get_scientific_name(tax_id, scientific_names_from_release_table)
+        for asm in tax_all_supported_asm[tax_id]:
+            sources = get_assembly_sources(asm, eva_assemblies, dbsnp_assemblies)
+            insert_entry_into_db(private_config_xml_file, sources, tax_id, scientific_name, asm, diff_asm['source'],
+                                 remapping_version, release_version, -1, -1, "'{\"ALL\"}'")
```

Comment:
For me it makes sense for `num_studies` and `num_ss_ids` to be filled in this script as well... if you'd prefer it done in a follow-up ticket that's fine, we just need to not forget about it :sweat_smile:

Reply:
Isn't it something that should be manage within [EVA-2946](https://www.ebi.ac.uk/panda/jira/browse/EVA-2946)?

Reply:
I thought this PR was covering EVA-2946 as well, as it fills in the studies list... (line 68 below).  Maybe there's another PR for that ticket but I couldn't find it.

Reply:
You're probably right, I have not come around to review it yet.

Reply:
added num_ss_ids counts logic